core/utils/mailbox.go

```diff
@@ -20,34 +20,34 @@ type Mailbox struct {
 
 // NewHighCapacityMailbox create a new mailbox with a capacity
 // that is better able to handle e.g. large log replays
-func NewHighCapacityMailbox() *Mailbox {
-	return NewMailbox(100000)
+func NewHighCapacityMailbox[T any]() *Mailbox[T] {
+	return NewMailbox[T](100000)
 }
 
 // NewMailbox creates a new mailbox instance
-func NewMailbox(capacity uint64) *Mailbox {
+func NewMailbox[T any](capacity uint64) *Mailbox[T] {
 	queueCap := capacity
 	if queueCap == 0 {
 		queueCap = 100
 	}
-	return &Mailbox{
-		chNotify: make(chan struct{}, 1),
-		queue:    make([]interface{}, 0, queueCap),
-		capacity: capacity,
-	}
+	var m Mailbox[T]
```

Comment:
Interesting, no generic struct literal construction syntax sugar yet?

Reply:
It may have been that I couldn't figure it out at the time, or that something changed since. Will definitely revisit. :+1:

Reply:
Fixed